pkg/awsutils/awsutils.go

```diff
@@ -1377,7 +1371,7 @@ func (cache *EC2InstanceMetadataCache) waitForENIAndIPsAttached(eni string, want
 				// At least some are attached
 				eniMetadata = returnedENI
 				// ipsToAllocate will be at most 1 less then the IP limit for the ENI because of the primary IP
-				if (eniIPCount > wantedSecondaryIPsOrPrefixes && !cache.useIPv4PrefixDelegation) || (eniIPCount >= wantedSecondaryIPsOrPrefixes && cache.useIPv4PrefixDelegation) {
+				if (eniIPCount > wantedSecondaryIPsOrPrefixes && !cache.enableIpv4PrefixDelegation) || (eniIPCount >= wantedSecondaryIPsOrPrefixes && cache.enableIpv4PrefixDelegation) {
```

Comment:
So the diff between PD case and secondary IP case ( > and >=) is due to the presence of Primary IP in secondary IP mode? But, we don't use the Primary IP for Pods anyways, isn't it? Can you elaborate a bit on it here?

Reply:
Yes the difference is having Primary IP in `returnedENI.IPv4Addresses` and wantedSecondaryIPsOrPrefixes will have maxIPsPerENI in secondary Mode which will not include primary IP. . But having said that because of the recent fix with warm IP targets, we need to check `eniIPCount >= wantedSecondaryIPsOrPrefixes` for both cases since eniIPCount has already removed primary IP and wantedSecondaryIPsOrPrefixes will either have max IPs or IPs based on warm targets.

Reply:
Ok, then probably we can remove the PD check and combine them in to one check

Reply:
Yeah will be updating the PR.